packages/serverpod/lib/src/server/websocket_request_handlers/method_websocket_request_handler.dart

```diff
@@ -322,6 +333,88 @@ class _MethodStreamManager {
     );
   }
 
+  void _handleMethodStreamEndpoint(
+    MethodStreamConnector methodConnector,
+    Session session,
+    Map<String, dynamic> args,
+    OpenMethodStreamCommand message,
+    Server server,
+  ) {
+    methodConnector.call(session, args).listen(
+      (value) {
+        _postMessage(
+          endpoint: message.endpoint,
+          method: message.method,
+          connectionId: message.connectionId,
+          message: MethodStreamMessage.buildMessage(
+            endpoint: message.endpoint,
+            method: message.method,
+            connectionId: message.connectionId,
+            object: server.serializationManager.encodeWithType(value),
+          ),
+        );
+      },
+      onDone: () async {
+        _postMessage(
+          endpoint: message.endpoint,
+          method: message.method,
+          connectionId: message.connectionId,
+          message: CloseMethodStreamCommand.buildMessage(
+            endpoint: message.endpoint,
+            connectionId: message.connectionId,
+            method: message.method,
+            reason: CloseReason.done,
+          ),
+        );
```

Comment:
Question: Would it be possible to finalize the build message inside of the postMessage? This way we would not have to duplicate the input params.

Reply:
I understand the "bloat" feeling but since we post different messages in multiple places and the buildMessage method is not the some between all messages I think it might be more clear to be explicit about what message we are sending.